src/Microsoft.Azure.WebJobs.Host/Extensions/IConfigurationExtensions.cs

```diff
@@ -93,5 +93,37 @@ public static bool IsSettingEnabled(this IConfiguration configuration, string se
             
             return false;
         }
+
+        public static IConfigurationSection GetWebJobsConnectionStringSection(this IConfiguration configuration, string connectionStringName)
+        {
+            // first try prefixing
+            string prefixedConnectionStringName = GetPrefixedConnectionStringName(connectionStringName);
+            IConfigurationSection section = GetConnectionStringOrSettingSection(configuration, prefixedConnectionStringName);
+
+            if (!section.Exists())
+            {
+                // next try a direct unprefixed lookup
+                section = GetConnectionStringOrSettingSection(configuration, connectionStringName);
+            }
+
+            return section;
+        }
+
+        /// <summary>
+        /// Looks for a connection string by first checking the ConfigurationStrings section, and then the root.
+        /// </summary>
+        /// <param name="configuration">The configuration.</param>
+        /// <param name="connectionName">The connection string key.</param>
+        /// <returns></returns>
+        public static IConfigurationSection GetConnectionStringOrSettingSection(this IConfiguration configuration, string connectionName)
```

Comment:
Doesn't need to be public right, since it appears it's only used by the method above?

Reply:
right. I'll make it private